### ISSUE TITLE
Resources index bug / filter bug

### ISSUE THREAD

**Issue by orlyapps** [laravel/nova-issues#1129]
Laravel Version: 5.7.17
Nova Version: 1.2.1
PHP Version: 7.2

### Description:
Index from any resource calls two times or more. The first api call get all resources. The Second call is my filter call. The first api call response 20 items and takes longer as my second filter call with 3 items. So first i see my filtered list and 1 second later the 20 items. It looks like the first api is too slow and overwrites the items.

If i add an sleep(2) to my filter it works :) Cause then my filter call takes longer then the first call ...

### Steps To Reproduce:
1. Create a Select Filter with some options
2. Add a default method and return a value wich delimit the items
3. Seed more then 500 resources
4. Click on the resource index

My Filter:
```
class ReorderStatus extends Filter
{
    /**
     * The filter's component.
     *
     * @var string
     */
    public $component = 'select-filter';

    /**
     * Apply the filter to the given query.
     *
     * @param  \Illuminate\Http\Request  $request
     * @param  \Illuminate\Database\Eloquent\Builder  $query
     * @param  mixed  $value
     * @return \Illuminate\Database\Eloquent\Builder
     */
    public function apply(Request $request, $query, $value)
    {
        // sleep(2) // Workaround for the filter bug
        return $query->whereStatus($value);
    }

    public function default()
    {
        return 'draft';
    }    

    /**
     * Get the filter's available options.
     *
     * @param  \Illuminate\Http\Request  $request
     * @return array
     */
    public function options(Request $request)
    {
        return [
            'Müssen nachbestellt werden' => 'draft',
            'Wurden bereits nachbestellt' => 'ordered',
            'Abgeschlossene' => 'completed'
        ];
    }
}

```

![resourcesindexbug 1](https://user-images.githubusercontent.com/5220826/49992894-988e5c80-ff85-11e8-937e-0d62e63f2dd0.gif)

![bildschirmfoto 2018-12-14 um 09 30 52](https://user-images.githubusercontent.com/5220826/49992762-3e8d9700-ff85-11e8-8fd5-3601a14b629d.png)

**Comment by dkulyk** [comment]
I confirm that there is such a bug

**Comment by dkulyk** [comment]
@davidhemphill I think need to request filters before fetching of resources in initialization.

**Comment by noahlocke** [comment]
I can also confirm. @davidhemphill is there a fix planned for this?

**Comment by dkulyk** [comment]
Check it on 1.3.0.

**Comment by aurawindsurfing** [comment]
I'm on 1.3.2 and I still see the same behaviour. Was that fixed in 1.3.0 @dkulyk @dillingham ?

**Comment by aurawindsurfing** [comment]
I also reported the same behaviour here: https://github.com/laravel/nova-issues/issues/848#issuecomment-443369832
